gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageWriteChannel.java

```diff
@@ -20,11 +20,11 @@
 @ThreadSafe
 final class CloudStorageWriteChannel implements SeekableByteChannel {
 
-  private GcsOutputChannel channel;
+  private WriteChannel channel;
```

Comment:
> Truncate is how I thought the GCS API works by default

I am not sure what you mean by that. In GCS Blobs are immutable.

Writing content with an existing blob name seemingly "override" the content.
Technically this is not a true override as it creates a new version of blob which
will be considered as "latest".

gcloud-java-storage is based on the apiary/api-client JSON Api (and not gRPC which is not yet available).

I am not familiar with any dedicate truncate operation, which I thought is what you were referring to.

A blob is only available for read and listing once it was finalized (using the close method). So, yes you can discard all changes by discarding the previously opened channel and create a new one
with the same name but that is not the same as truncate.